modules/loadbalancer/main.tf

```diff
@@ -61,106 +58,148 @@ resource "azurerm_lb" "lb" {
+      subnet_id                     = frontend_ip.value.subnet_id
+      private_ip_address_allocation = frontend_ip.value.private_ip_address != null ? "Static" : null
+      private_ip_address            = frontend_ip.value.private_ip_address
+      zones                         = frontend_ip.value.subnet_id != null ? var.zones : null
+
+      gateway_load_balancer_frontend_ip_configuration_id = frontend_ip.value.gwlb_fip_id
+    }
+  }
+
+  lifecycle {
+    precondition {
+      condition = !(
+        anytrue(
+          [for _, fip in var.frontend_ips : fip.public_ip_name != null]
+          ) && anytrue(
+          [for _, fip in var.frontend_ips : fip.subnet_id != null]
+        )
+      )
+      error_message = "All frontends have to be of the same type, either public or private. Please check module's documentation (Usage section) for details."
     }
   }
 }
 
-resource "azurerm_lb_backend_address_pool" "lb_backend" {
+resource "azurerm_lb_backend_address_pool" "this" {
   name            = var.backend_name
-  loadbalancer_id = azurerm_lb.lb.id
+  loadbalancer_id = azurerm_lb.this.id
 }
 
-resource "azurerm_lb_probe" "probe" {
-  name            = var.probe_name
-  loadbalancer_id = azurerm_lb.lb.id
-  port            = var.probe_port
+locals {
+  default_http_probe_port = {
+    "Http"  = 80
+    "Https" = "443"
+  }
+  default_probe = (
+    var.health_probes == null || anytrue(flatten([
+      for _, fip in var.frontend_ips : [
+        for _, rule in fip.in_rules : rule.health_probe_key == "default"
+      ]
+    ]))
+    ) ? {
+    default = {
+      name                = "default_vmseries_probe"
+      protocol            = "Tcp"
+      port                = 80
+      probe_threshold     = null
+      interval_in_seconds = null
+      request_path        = null
+    }
+  } : {}
 }
 
-resource "azurerm_lb_rule" "in_rules" {
+resource "azurerm_lb_probe" "this" {
+  for_each = merge(coalesce(var.health_probes, {}), local.default_probe)
+
+  loadbalancer_id = azurerm_lb.this.id
+
+  name                = each.value.name
+  protocol            = each.value.protocol
+  port                = contains(["Http", "Https"], each.value.protocol) && each.value.port == null ? local.default_http_probe_port[each.value.protocol] : each.value.port
+  probe_threshold     = each.value.probe_threshold
+  interval_in_seconds = each.value.interval_in_seconds
+  request_path        = each.value.protocol != "Tcp" ? each.value.request_path : null
+
+  # this is to overcome the discrepancy between the provider and Azure defaults
+  # for more details see here -> https://learn.microsoft.com/en-gb/azure/load-balancer/whats-new#known-issues:~:text=SNAT%20port%20exhaustion-,numberOfProbes,-%2C%20%22Unhealthy%20threshold%22
+  number_of_probes = 1
+}
+
+resource "azurerm_lb_rule" "this" {
```

Comment:
Currently we have:
- `azurerm_lb_rule`
- `azurerm_lb_outbound_rule`

but we are missing `azurerm_lb_nat_rule` - what do you thing about adding support for that ?

Reply:
we can, if there is a use case for it, with NGFW that is